SpatialGDK/Source/SpatialGDKEditorCommandlet/Private/Commandlets/GenerateSchemaAndSnapshotsCommandlet.cpp

```diff
@@ -22,75 +23,146 @@ int32 UGenerateSchemaAndSnapshotsCommandlet::Main(const FString& Args)
 {
 	UE_LOG(LogSpatialGDKEditorCommandlet, Display, TEXT("Schema & Snapshot Generation Commandlet Started"));
 
-	//NOTE: For future use, if schema generation configuration at the command line is desired
-	//TArray<FString> Tokens;
-	//TArray<FString> Switches;
-	//TMap<FString, FString> Params;
-	//ParseCommandLine(*Args, Tokens, Switches, Params);
+	TArray<FString> Tokens;
+	TArray<FString> Switches;
+	TMap<FString, FString> Params;
+	ParseCommandLine(*Args, Tokens, Switches, Params);
+
+	//TODO: Optionally clean up previous schema and snapshot files once the UnrealGDK provides support to do so
+	GeneratedMapPaths.Empty();
 
 	FSpatialGDKEditor SpatialGDKEditor;
-	GenerateSchema(SpatialGDKEditor);
-	GenerateSnapshots(SpatialGDKEditor);
+	if (Params.Contains(MapPathsParamName))
+	{
+		FString MapNameParam = *Params.Find(MapPathsParamName);
+
+		//Spaces are disallowed in all paths, so we just check for them now and exit early if an invalid path was provided
+		if (MapNameParam.Contains(TEXT(" ")))
+		{
+			UE_LOG(LogSpatialGDKEditorCommandlet, Error, TEXT("%s argument may not contain spaces."), *MapPathsParamName);
+				return 1;
+		}
+
+		FString ThisMapName;
+		FString RemainingMapPaths = MapNameParam;
+		while (RemainingMapPaths.Split(TEXT(";"), &ThisMapName, &RemainingMapPaths))
```

Comment:
same thing here for text macro but it's up to you. I don't mind if you'd prefer it.

Reply:
I've opted to keep TEXT macro usage in for all string literal usages, as per Epic's recommendations.